Include/cpython/pystate.h

```diff
@@ -204,6 +204,9 @@ struct _ts {
 #  else
 #    define Py_C_RECURSION_LIMIT 500
 #  endif
+#elif defined(Py_COVERAGE)
+   // With gcov, the stack usage is higher.
+#  define Py_C_RECURSION_LIMIT 500
```

Comment:
Hmm, why don't merge this with the Py_DEBUG ifdef on L198?  The reason is exactly same, it's common for all non-optimized builds (``-O0``, ``-Og``, ``-O``).

Maybe we should instead define some macro, that will be enabled for all non-optimized builds.  With this you could also simplify the ``python_is_optimized()`` helper.

Reply:
That would be a larger change, which makes some sense but also requires plenty of work for the make file - we need to pick out all the recipes that is not optimized and add a macro to it.

Sure this can be merged with the `Py_DEBUG` line, but this way keeps some flexibility. All the other cases are actually similar as well - they need extra stack space. I don't have a very strong preference on this, leaning slightly toward separate cases.